src/main/java/lab3/ui/terminalConsole.java

```diff
@@ -0,0 +1,104 @@
+     * @param board the tic tac toe board
+     */
+    public terminalConsole(Board board) {
+        this.board = board;
+    }
+
+    /**
+     * Displays the current board state
+     */
+    public void displayBoard() {
+        System.out.println(board);
+    }
+
+    /**
+     * Prints a message to the console
+     * @param message the message to be printed
+     */
+    public static void println(String message) {
+        System.out.println(message);
+    }
+
+    /**
+     * Displays a message if a player has reached a win state
+     */
+    public static void displayWin() {
+        println("PLAYER " + currentPlayerSymbol() + " HAS WON");
+        println(String.format("THE SCORE IS NOW %s (X) VS %s (O)", callScore().XWins(), callScore().OWins()));
+    }
+
+    /**
+     * Displays a tie message if neither player was able to reach a win state
+     */
+    public static void displayTie() {
+        println("TIE GAME, NO ONE WINS!");
+        println(String.format("THE SCORE REMAINS %s (X) VS %s (O)", callScore().XWins(), callScore().OWins()));
+    }
+
+    /**
+     * Prompts both players if they would like to play again
+     * @return true if both players want to play again, false if not
+     */
+    public boolean rematch() {
+        Scanner scanner = new Scanner(System.in);
+        println("Do both players wish to play again? (y/n)");
+        while (true) {
+            println("Player X?");
+            String playerX = scanner.nextLine().replaceAll(" ", "").toLowerCase();
+            if (!playerX.equals("y") && !playerX.equals("n")) {
+                println("please enter Y or N to continue.");
+            } else {
+                while (true) {
+                    println("Player O?");
+                    String playerO = scanner.nextLine().replaceAll(" ", "").toLowerCase();
+                    if (!playerO.equals("y") && !playerO.equals("n")) {
+                        println("please enter Y or N to continue.");
+                    } else {
+                        return playerX.equals("y") && playerO.equals("y");
+                    }
+                }
+            }
+        }
+    }
+    /**
+     * Prompts the current player to place a move on the board
+     */
+    public void move() {
+        Scanner scanner = new Scanner(System.in);
+
+        while (true) {
+            println("Player " + currentPlayerSymbol() + ": Please Select a Square: ");
+            String input = scanner.nextLine().replaceAll(" ", "").toUpperCase();
+
+            try {
+                Coordinates coord = Coordinates.valueOf(input);
+                if (board.setMove(coord)) {
+                    return;
+                } else {
+                    println("Tile not available. Please enter new coordinates.");
+                }
+            } catch (IllegalArgumentException e) {
+                println("Invalid coordinate. Please enter a valid move (e.g., A1, B2, C3).");
+            }
+        }
+    }
```

Comment:
My hope for your UI package was that there would be very little game related logic here, and mostly class methods to which you pass information that needs to be displayed.  Here, you're mixing your game logic (the actual call to `setMove` with your UI logic (the obtaining of which Coordinate the user chose).  See the sample solution and compare.  In the sample, the UI simply returns the position selected by the user and doesn't need to know anything about the current state of the game, or how that position will be used.  The game logic takes that value and updates the game accordingly